source/common/protobuf/utility.h

```diff
@@ -65,7 +65,7 @@ uint64_t fractionalPercentDenominatorToInt(const envoy::type::FractionalPercent&
 // @param default_value supplies the default if the field is not present.
 #define PROTOBUF_PERCENT_TO_ROUNDED_INTEGER_OR_DEFAULT(message, field_name, max_value,             \
                                                        default_value)                              \
-  ((message).has_##field_name()                                                                    \
+  ((message).has_##field_name() && !std::isnan((message).field_name().value())                     \
        ? ProtobufPercentHelper::convertPercent((message).field_name().value(), max_value)          \
```

Comment:
I _think_ this is fine to mitigate the issue today. The clean way to do this is to catch NaN in PGV annotations (i.e. https://github.com/envoyproxy/envoy/blob/master/api/envoy/type/percent.proto#L14, https://github.com/lyft/protoc-gen-validate/blob/master/validate/validate.proto#L98), the same way we validate the [0..100] range.

Can you file an issue for NaN validation at https://github.com/lyft/protoc-gen-validate and add a TODO comment here?

@rodaine @akonradi does this make sense to you?

Reply:
Agreed, this would be a good validation to have in PGV - it's simple, self-contained, and generally useful.

Reply:
Done both 👍

Reply:
Thanks for filing the issue. I think you need a small change of behavior here. Instead of returning the default in NaN, it should `throw EnvoyException()`, to allow the NaN to be reflected back to the config pipeline.